### DISCUSSION TITLE
Improvements to the original translatable English strings.

### DISCUSSION THREAD

**Discussion by hboetes** [nicotine-plus/nicotine-plus#1834] [Ideas]
In this thread, we discuss suggestions to improve the original English strings. Especially the translators, who read all the original strings, find expressions that can be improved upon.

**Comment by hboetes** [comment]
From `pynicotine/__init__.py line 39`, part of the output of `./nicotine -h`
```
  -u dir, --user-data dir
                use non-default user data directory for e.g. list of
                downloads
```
This is the `~/.local/share/nicotine/` folder, which contains all the application data; shares DB, download/upload list, logs etc.

Proposed string(pseudo code), with the additional benefit of showing the default folder, so people can inspect it:
``` 
use custom application data folder, default is %(default_datadir)
```

**Reply by slook** [reply]
> the additional benefit of showing the default folder

This would indeed be nice, but it is not usual for a console help message to display dynamic data. It is possible that some unforeseen error could cause the program to crash instead of displaying the help message on stdout., therefore only static content is allowable here.

**Reply by hboetes** [reply]
> > the additional benefit of showing the default folder
> 
> This would indeed be nice, but it is not usual for a console help message to display dynamic data. It is possible that some unforeseen error could cause the program to crash instead of displaying the help message on stdout., therefore only static content is allowable here.

Do you have a reference on that?

**Reply by slook** [reply]
Not specifically no, just try any command with `--help` and all of them contain only static text, for example...

`$ man --help`
```
Usage: man [OPTION...] [SECTION] PAGE...

  -C, --config-file=FILE     use this user configuration file
...
```
It is not possible to see the default '`FILE`' used for `--config-file`, to find that out you have to look in the manual page for `man`...

`$ man man`
```
...
FILES
       /etc/manpath.config
              man-db configuration file.

       /usr/share/man
              A global manual page hierarchy.

       /usr/share/man/index.(bt|db|dir|pag)
              A traditional global index database cache.

       /var/cache/man/index.(bt|db|dir|pag)
              An FHS compliant global index database cache.

```
... but this static info is always pulled from the `man-db`, and does not interact with binary executables to pull dynamic data.

The `man` page for `nicotine` does also contain default config information in the "FILES" section of its `man` page... 
`$ man nicotine`

```
NICOTINE+(1)                       General Commands Manual                       NICOTINE+(1)

NAME
       nicotine - graphical client for the Soulseek peer-to-peer network

SYNOPSIS
       nicotine [options]

DESCRIPTION
       Nicotine+  is  a graphical application that allows users to interact with the Soulseek
       peer-to-peer network.

OPTIONS
       -c <file>, --config=<file>
              Use <file> as configuration file. If <file> does not exists a new empty config‐
              uration file will be created.

       -u <dir>, --user-data=<dir>
              Use  <dir> as directory for user data, e.g. list of shared files and downloads.
              If <dir> does not exists a new empty directory will be created.

...
...

FILES
       ~/.config/nicotine/config
              The configuration file.

       ~/.local/share/nicotine/
              Other personal files such as downloads, shares and logs.

AUTHOR
       Nicotine+ Team <https://nicotine-plus.org/>

                                                                                 NICOTINE+(1)
 Manual page nicotine(1) line 21/68 (END) (press h for help or q to quit)
```

**Comment by hboetes** [comment]
Now that we're on it:

From `pynicotine/__init__.py` line 34, part of the output of `./nicotine -h`

```
  -c file, --config file
                        use non-default configuration file
```

```
use alternative configuration file, default is $(default_config_file)
```

**Reply by slook** [reply]
See above https://github.com/nicotine-plus/nicotine-plus/discussions/1834#discussioncomment-1984995

**Reply by slook** [reply]
To match the format notation of the built-in [argparse.DefaultHelpFormatter](https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter) the following syntax could be used:

`        help=_("use non-default configuration file") + (" (default: %s)" % config.filename`)

_This prints output as follows:_
```
  -c file, --config file
                        use non-default configuration file (default: /home/user/.nicotine/config)
```

----

However, frankly I feel the 3 options (-c, -u, -p) for changing directories is excessive, and could be more useful if everything was combined into one option to simply control the selection of the entire profile folder, unless of course there is explained a specific use case for having the fine-grained control currently present?

**Reply by hboetes** [reply]
I highly doubt there is even one n+ user that actually uses these options regularly, but I can imagine of use cases for them, especially for development and testing purposes. They don't annoy me at all. Perhaps the -h doesn't need to include them.

**Reply by slook** [reply]
It is easy to see why these probably aren't very popular options, because it's frustrating that setting up N+ to be used with a fresh config/database/plugins (as in for an alternate user account) require specifying all three `-c` `-u` `-p` options with full paths, which results in a very large command!

Also whilst I'm thinking about pruning the command line options, how do you feel about enabling (`-t`) and disabling (`-d`) the tray icon from the command line, is this really needed?

**Reply by mathiascode** [reply]
Almost all of these were added before I started working on Nicotine+. The only reason we have three options, is because two of them already existed, and I didn't want to break compatibility.

I wouldn't mind combining them into one option, and possibly removing tray options, but this needs to be mentioned in the release notes.

**Reply by mathiascode** [reply]
Actually, it might be a good idea two keep two options for config file and user data, since the XDG base directory spec splits them, but the plugin directory option is probably redundant now.

**Reply by slook** [reply]
> it might be a good idea two keep two options for config file and user data

Yes of course, the `--config` and `--user-data` and `--plugins` arguments should always be kept, if at least as hidden arguments.

> the plugin directory option is probably redundant ("`-p`")

In that case, I propose that the `-p` option in future becomes `--profile` (to start with, we can add a hidden argument as an experimental feature). This seems to be a common standardized option for an easy way of specifying a single `dir` in which to store an application's config+user-data+plugins together all in one location that is easy for user to locate and back-up etc.

**Comment by hboetes** [comment]
I'm a bit more careful, but on the other hand I see no use for these options. @mathiascode what do you think they could be useful for?

**Reply by mathiascode** [reply]
Running multiple instances of Nicotine+.

**Reply by slook** [reply]
This is done by specifying an alternate `--config` file (a user-facing command line option), but in reality running multiple instances of Nicotine+ is problematic due to conflicting shares database issues, unless an alternate `--user-data` directory is also specified (but an end-user would not know this). Also, if plugins wanted to store data then conflicts could arise there also.

Considering the unpredictable nature of combining different config files and user-data dirs, it makes me wonder if we should restrict `multi_instance = True if (args.config and args.user_data) else False` such would help to avoid erroneous runtime behaviour.

**Comment by slook** [comment]
**None of the strings in the Spamfilter plugin make any sense**

One day the core plugin strings might be made translatable, but currently some strings in the plugins are badly written so they require review.

These ones are particularly confusing:

`'The minimum length of a line before it\'s considered as ASCII spam'`

`'The maximum number of different characters that is still considered ASCII spam'`

`'The maximum length of a line before it\'s considered as spam.'`

`'A line should be at least %s long with a maximum of %s different characters before it\'s considered ASCII spam.'`

From the end user's point of view, there is a lot of oppositional phrases and double-negatives.

**Reply by mathiascode** [reply]
Every plugin should be reviewed. I added support for translations in core plugins, but it's not applied yet because the strings need review (some have not been changed since the plugins were added years ago).

**Reply by slook** [reply]
Can the plugin settings have tooltips?

I feel that some of the labels are excessively long because some of the more complex settings require a deep explanation.

**Comment by hboetes** [comment]
I propose changing:
> Wishlist items are auto-searched at regular intervals, for discovering uncommon files.

With:

> Wishlist items are periodically searched, to help discovering uncommon files.

Or:

> Wishlist items are searched regularly, to help you find rare files.

**Reply by slook** [reply]
> Wishlist items are searched regularly to help find rare files.

**Comment by hboetes** [comment]
I propose changing:

> Enter an IP address you want to ignore:
> Enter an IP address you want to ban:

With:

> Enter the IP address you want to ignore:
> Enter the IP address you want to ban:

**Reply by slook** [reply]
> IP address to ignore:
> IP address to ban:

**Comment by hboetes** [comment]
I'd like to propose changing:

> Show reverse file paths (requires a restart)

with:

> Show file path in reverse order (requires a restart)

**Reply by slook** [reply]
> Reverse folder paths (requires a restart)

This change could wait until we engineer a method to avoid the need for restarting the application.

It might be sensible to create a separate string for "(requires a restart)" or "(restart required)" to allow us flexibility to apply/remove this note to/from any config label, and reduce duplicate translation effort.

**Reply by hboetes** [reply]
Makes sense, but why the thumbs down?

**Reply by slook** [reply]
I don't like your suggestion because it's overly long-winded. Also the word "order" is ambiguous with sort order.

**Reply by hboetes** [reply]
The sentence will use 50% of the available space, and for the mere price of two extra words, 7 extra chars, it explains much clearer what the function does.

**Reply by slook** [reply]
The inclusion of the word "Show" wrongly implies that some element is entirely hidden if the option is off. Mentioning "file" here is misleading, because there is no filename in the displayed path, which is why I suggest changing that word to "folder" instead. How about:

> "Reversed folder paths"

**Reply by hboetes** [reply]
I can agree with that. Terse and explains exactly what it does. Thanks for your suggestion.

**Comment by hboetes** [comment]
I'd like to propose changing

> Allow users to send you any files:

With:

> Allow users to send you files:

**Reply by mathiascode** [reply]
"any" was a deliberate choice to highlight that users can send/upload anything they want if the option is enabled. We could perhaps reword it if there are better suggestions, but I'm afraid that being too generic here would make it easier to enable by accident.

**Reply by hboetes** [reply]
In that case, 
> Allow users to send you any file:

Would be better English.

**Reply by slook** [reply]
"files" is a valid plural in this context. Leave as is.

**Comment by hboetes** [comment]
I'd like to propose changing:
> Search result filters are used to refine which search results are displayed.

With:
> To narrow down the search results, you can use search result filters.

**Reply by slook** [reply]
> Result filters narrow down the search results.

**Comment by mathiascode** [comment]
Suggested by ketacat in the IRC channel:

> Add User Note

 ->

> Edit User Note

**Reply by slook** [reply]
`_("Edit User Note") if user.note else _("Add User Note")`
